sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -506,12 +511,71 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         AttributeReference("value", StringType)()), true)
     }
 
+    val namedExpressions = expressions.map {
+      case e: NamedExpression => e
+      case e: Expression => UnresolvedAlias(e)
+    }
```

Comment:
re discussion in https://github.com/apache/spark/pull/25028#issuecomment-547230458 and https://github.com/apache/spark/pull/25028#issuecomment-547246714 and 

Alternative suggestion:
```
   val aliasFunc = (position: Int, e: Expression) => s"gen_alias_${position}"

    val namedExpressions = expressions.zipWithIndex.map {
      case (e: NamedExpression, _) => e
      case (e: Expression, index) => UnresolvedAlias(e, Some(aliasFunc(index, _)))
    }
```

Reply:
It is ok even to rename all expression with `aliasFunc`  in this place.
After current busy job I will restart this pr and follow your nice suggestion.